crates/driver/src/boundary/liquidity/swapr.rs

```diff
@@ -0,0 +1,55 @@
+use {
+    crate::{
+        boundary,
+        domain::liquidity::{self, swapr},
+        infra::{self, blockchain::Ethereum},
+    },
+    anyhow::Result,
+    shared::{
+        current_block::CurrentBlockStream,
+        price_estimation,
+        sources::{swapr::reader::SwaprPoolReader, uniswap_v2::pool_fetching::DefaultPoolReader},
+    },
+    solver::{liquidity::ConstantProductOrder, liquidity_collector::LiquidityCollecting},
+};
+
+/// The base unit for basis points, i.e. how many basis points in 100%.
+const BPS_BASE: u32 = 10_000;
+
+pub fn to_domain(id: liquidity::Id, pool: ConstantProductOrder) -> Option<liquidity::Liquidity> {
+    assert_eq!(
+        (pool.fee.numer() * BPS_BASE) % pool.fee.denom(),
+        0,
+        "invalid Swapr fee ratio; does not have exact BPS representation",
+    );
```

Comment:
This will panic. Is that what you want?

Reply:
I guess I can make it s `debug_assert_eq`. My rationale for `panic`-ing here is that it should really only happen if there is a configuration issue.

Now that I think about it more, a misbehaving node could return an invalid result from the `swapFee` and cause us to panic... I will remove it.

Reply:
Removed panic.